GenerateHWIntrinsicTests_Arm/project.assets.json already exists

Looks like some sort of race during restore of the generated HW intrinsics tests. Cc @tannergooding 

```
2022-12-20T04:03:50.9484776Z /__w/1/s/.dotnet/sdk/7.0.100/NuGet.RestoreEx.targets(19,5): error : The file '/__w/1/s/artifacts/tests/coreclr/packages/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm/project.assets.json' already exists. [/__w/1/s/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.csproj] [/__w/1/s/src/tests/JIT/HardwareIntrinsics/Arm/AdvSimd/AdvSimd_ro.csproj] [/__w/1/s/src/tests/build.proj]
2022-12-20T04:03:50.9636971Z ##[error].dotnet/sdk/7.0.100/NuGet.RestoreEx.targets(19,5): error : The file '/__w/1/s/artifacts/tests/coreclr/packages/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm/project.assets.json' already exists. [/__w/1/s/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.csproj] [/__w/1/s/src/tests/JIT/HardwareIntrinsics/Arm/AdvSimd/AdvSimd_ro.csproj]
```

## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=116668
Build error leg or test failing: Build / coreclr Common Pri0 Test Build AnyOS AnyCPU checked / Build managed test components
Pull request: https://github.com/dotnet/runtime/pull/79842

## Error Message
Fill the error message using [known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-message-section).

```json
{
  "ErrorMessage": "project.assets.json' already exists",
  "BuildRetry": false
}
```


<!--Known issue error report start -->
### Report
|Build|Definition|Step Name|Console log|Pull Request|
|---|---|---|---|---|
|[117626](https://dev.azure.com/dnceng-public/public/_build/results?buildId=117626)|dotnet/runtime|Build managed test components|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/117626/logs/572)|dotnet/runtime#79283|
|[117617](https://dev.azure.com/dnceng-public/public/_build/results?buildId=117617)|dotnet/runtime|Build managed test components|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/117617/logs/1046)||
|[117585](https://dev.azure.com/dnceng-public/public/_build/results?buildId=117585)|dotnet/runtime|Build managed test components|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/117585/logs/154)||
|[117323](https://dev.azure.com/dnceng-public/public/_build/results?buildId=117323)|dotnet/runtime|Build managed test components|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/117323/logs/421)||
|[117295](https://dev.azure.com/dnceng-public/public/_build/results?buildId=117295)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/117295/logs/1041)|dotnet/runtime#79339|
|[117286](https://dev.azure.com/dnceng-public/public/_build/results?buildId=117286)|dotnet/runtime|Build managed test components|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/117286/logs/1127)|dotnet/runtime#79416|
|[117202](https://dev.azure.com/dnceng-public/public/_build/results?buildId=117202)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/117202/logs/1785)|dotnet/runtime#77304|
|[117084](https://dev.azure.com/dnceng-public/public/_build/results?buildId=117084)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/117084/logs/1681)|dotnet/runtime#79046|
|[116803](https://dev.azure.com/dnceng-public/public/_build/results?buildId=116803)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/116803/logs/1041)|dotnet/runtime#79849|
|[116788](https://dev.azure.com/dnceng-public/public/_build/results?buildId=116788)|dotnet/runtime|Build managed test components|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/116788/logs/919)|dotnet/runtime#79450|
|[116777](https://dev.azure.com/dnceng-public/public/_build/results?buildId=116777)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/116777/logs/637)||
|[116768](https://dev.azure.com/dnceng-public/public/_build/results?buildId=116768)|dotnet/runtime|Build managed test components|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/116768/logs/330)|dotnet/runtime#79847|
|[116713](https://dev.azure.com/dnceng-public/public/_build/results?buildId=116713)|dotnet/runtime|Build managed test components|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/116713/logs/125)||
|[116668](https://dev.azure.com/dnceng-public/public/_build/results?buildId=116668)|dotnet/runtime|Build managed test components|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/116668/logs/484)|dotnet/runtime#79842|
|[116550](https://dev.azure.com/dnceng-public/public/_build/results?buildId=116550)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/116550/logs/1681)|dotnet/runtime#79721|
|[116496](https://dev.azure.com/dnceng-public/public/_build/results?buildId=116496)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/116496/logs/1653)|dotnet/runtime#79169|
|[116489](https://dev.azure.com/dnceng-public/public/_build/results?buildId=116489)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/116489/logs/1000)|dotnet/runtime#79834|
|[116457](https://dev.azure.com/dnceng-public/public/_build/results?buildId=116457)|dotnet/runtime|Build managed test components|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/116457/logs/712)|dotnet/runtime#79685|
|[116443](https://dev.azure.com/dnceng-public/public/_build/results?buildId=116443)|dotnet/runtime|Build managed test components|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/116443/logs/628)|dotnet/runtime#79831|
|[116369](https://dev.azure.com/dnceng-public/public/_build/results?buildId=116369)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/116369/logs/1720)|dotnet/runtime#79416|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|7|20|20|
<!--Known issue error report end -->

--- a/src/tests/JIT/HardwareIntrinsics/Arm/Directory.Build.targets
+++ b/src/tests/JIT/HardwareIntrinsics/Arm/Directory.Build.targets
@@ -12,7 +12,7 @@
           Inputs="$(MSBuildAllProjects);$(RepoRoot)/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs"
           Outputs="$(GeneratedHWIntrinsicTestListFile)"
           Condition="'$(Language)' == 'C#'">
-    <Exec Command="$(DotNetCli) build $(RepoRoot)/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.csproj -c $(Configuration) /p:IntermediateOutputPath=$(IntermediateOutputPath)/GenerateHWIntrinsicTests/ /p:OutputPath=$(OutputPath)/GenerateHWIntrinsicTests/" />
+    <Exec Command="$(DotNetCli) build $(RepoRoot)/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.csproj -c $(Configuration) --no-restore /p:IntermediateOutputPath=$(IntermediateOutputPath)/GenerateHWIntrinsicTests/ /p:OutputPath=$(OutputPath)/GenerateHWIntrinsicTests/" />
     <Exec Command="$(DotNetCli) exec $(OutputPath)/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.dll $(MSBuildProjectName) $(MSBuildThisFileDirectory)Shared $(GeneratedHWIntrinsicTestDirectory) $(GeneratedHWIntrinsicTestListFile)" />
   </Target>
 

--- a/src/tests/JIT/HardwareIntrinsics/General/Directory.Build.targets
+++ b/src/tests/JIT/HardwareIntrinsics/General/Directory.Build.targets
@@ -12,7 +12,7 @@
           Inputs="$(MSBuildAllProjects);$(RepoRoot)/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_General.cs"
           Outputs="$(GeneratedHWIntrinsicTestListFile)"
           Condition="'$(Language)' == 'C#'">
-    <Exec Command="$(DotNetCli) build $(RepoRoot)/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_General.csproj -c $(Configuration) /p:IntermediateOutputPath=$(IntermediateOutputPath)/GenerateHWIntrinsicTests/ /p:OutputPath=$(OutputPath)/GenerateHWIntrinsicTests/" />
+    <Exec Command="$(DotNetCli) build $(RepoRoot)/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_General.csproj -c $(Configuration) --no-restore /p:IntermediateOutputPath=$(IntermediateOutputPath)/GenerateHWIntrinsicTests/ /p:OutputPath=$(OutputPath)/GenerateHWIntrinsicTests/" />
     <Exec Command="$(DotNetCli) exec $(OutputPath)/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_General.dll $(MSBuildProjectName) $(MSBuildThisFileDirectory)Shared $(GeneratedHWIntrinsicTestDirectory) $(GeneratedHWIntrinsicTestListFile)" />
   </Target>
 

--- a/src/tests/JIT/HardwareIntrinsics/X86/Directory.Build.targets
+++ b/src/tests/JIT/HardwareIntrinsics/X86/Directory.Build.targets
@@ -12,7 +12,7 @@
           Inputs="$(MSBuildAllProjects);$(RepoRoot)/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_X86.cs"
           Outputs="$(GeneratedHWIntrinsicTestListFile)"
           Condition="'$(Language)' == 'C#'">
-    <Exec Command="$(DotNetCli) build $(RepoRoot)/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_X86.csproj -c $(Configuration) /p:IntermediateOutputPath=$(IntermediateOutputPath)/GenerateHWIntrinsicTests/ /p:OutputPath=$(OutputPath)/GenerateHWIntrinsicTests/" />
+    <Exec Command="$(DotNetCli) build $(RepoRoot)/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_X86.csproj -c $(Configuration) --no-restore /p:IntermediateOutputPath=$(IntermediateOutputPath)/GenerateHWIntrinsicTests/ /p:OutputPath=$(OutputPath)/GenerateHWIntrinsicTests/" />
     <Exec Command="$(DotNetCli) exec $(OutputPath)/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_X86.dll $(MSBuildProjectName) $(MSBuildThisFileDirectory)Shared $(GeneratedHWIntrinsicTestDirectory) $(GeneratedHWIntrinsicTestListFile)" />
   </Target>
 

